ML Beetle info pages (#4)
Imageomics/HDR-SMood-Challenge: pages/data.md

# Beetles as Sentinel Taxa: Data

> This challenge uses images and metadata associated with carabid beetle specimens collected by the [National Ecological Observatory Network (NEON)](https://www.neonscience.org/) for the "Ground beetles sampled from pitfall traps" data product (DP1.10022.001, [RELEASE-2025](https://doi.org/10.48443/cd21-q875)) to predict drought conditions at the location and time of sampling. 

Reviewer: ```suggestion
This challenge uses images and metadata associated with [carabid beetle](https://en.wikipedia.org/wiki/Ground_beetle) specimens collected by the [National Ecological Observatory Network (NEON)](https://www.neonscience.org/) for the "Ground beetles sampled from pitfall traps" data product (DP1.10022.001, [RELEASE-2025](https://doi.org/10.48443/cd21-q875)) to predict drought conditions at the location and time of sampling. 
```

For this challenge, drought was characterized for each beetle sampling event using values of the Standardized Precipitation Evapotranspiration Index (SPEI) retrieved from the [GRIDMET Drought image collection](https://developers.google.com/earth-engine/datasets/catalog/GRIDMET_DROUGHT#description) (see [Abatzoglou (2011)](https://doi.org/10.1002/joc.3413).

> All data are licensed under <<get NEON preferred license infromation from Christine>>.

Reviewer: Presumably this needs to be filled in before merging, @sokole?

## General Overview of Inputs and Expected Outputs

**Model inputs** will be a flat metadata table (csv format) where each row is a metadata record for a specimen image. Multiple images will be associated with a sampling `event_id`, where a "sampleing event" corresponds with a location (NEON Site ID) and the date on which the beetle specimens were collected. Thus, multiple input records (multiple images of beetle specimens) will be used to predict a single SPEI value at a given site on a given date. Note that `event_id`s will be anonymized in the test dataset, but the metadata will include `site_id` and `collect_date` in the training data. 

**What to predict:** The SPEI metrics represent cumulative drought conditions over a time window. For this challenge, participants will submit predictions for SPEI metrics representing three different time scales: 

> - `spei_30d` is drought conditions calcualted from data for the 30 day window preceding sample collection

Reviewer: ```suggestion
- `spei_30d` is drought conditions calculated from data for the 30 day window preceding sample collection
```

- `spei_1y` is the drought condition calculated from the year preceding sample collection
- `spei_2y` is the drought condition calculated from the two years preceding sample collection

> **The submission file** will be a long-format flat table (csv file format) with a prediction for each `spei_30d`, `spei_1y`, and `spei_2y` for each `event_id`. Predictions will include a measure of uncertainty, so each prediction will include `mu` (mean) and `sigma` (standard devation). See the example submissoin below for more details. 

Reviewer: ```suggestion
**The submission file** will be a long-format flat table (csv file format) with a prediction for each `spei_30d`, `spei_1y`, and `spei_2y` for each `event_id`. Predictions will include a measure of uncertainty, so each prediction will include `mu` (mean) and `sigma` (standard deviation). See the example submission below for more details. 
```

## Instructions to Download Training Data

> <<needs updating for beetle ml process, once we have it developed>>

Reviewer: Presumably this also needs to be updated before merging, @sokole? Otherwise, we should choose a more appropriate (for public consumption) placeholder.

First, install the downloader in your virtual environment:
```bash
pip install git+https://github.com/Imageomics/cautious-robot
```

> Then download <<link to training data csv>> and run: 

Reviewer: Need to replace with actual link, @sokole?

Author: @hlapp this is information I don't have. Do we want to keep this PR open and have others make commits here, or should we merge this into the `update-pages` branch have others edit there?

Reviewer: I'm in principle good with either way. My main concern here is that we don't overlook one or more of these placeholders before It goes live. Does everyone else who needs to fill in information for the placeholders know that they do? Do we have some mechanism (whether a full issue or a checkbox somewhere) by which we're tracking completion?

So to pose the question slightly differently, if we ignore the presence of placeholders here and merge into `update-pages` as is, how will we know when `update-pages` is ready to go live and what's still missing to get there?

```bash
cautious-robot -i <path/to/beetle_ml_train.csv> -o <path/to/images> -s hybrid_stat -v md5
```

This will create subfolders ...

> <<are we going to provide a download of the images, or just a metadata file with image URIs?>>

Reviewer: Also needs to be updated before merging? @sokole

Author: same here. I don't have this info. Lisa or someone will need to update this section.

## Additional Information About the CSV File

Following the above steps, participants will obtain a csv metadata file with `image_uri`s linking to the specimen images. Each record (row) is metadata for a single specimen image. The training dataset will include all the fields described in the schema below. The test dataset will only include `event_id` (anonymized), `domain_id`, `scientific_name`, and `image_uri`.

> <<We can update this as appropriate if we are going to add metadata about the image, like image quality flags etc. Also, I'm unclear on whether we're going to provide a download of the image files to participants, or just give them a table with image_uris>>

Reviewer: Also needs updating? @sokole

Author: this will need to be updated by David? I don't know what the final format of the data will look like, or what additional metadata will be provided about the images.

Reviewer 2: This was updated and now points to the dataset repo.

# Beetles as Sentinel Taxa: Data

> This challenge uses images and metadata associated with [carabid beetle](https://en.wikipedia.org/wiki/Ground_beetle) specimens collected by the [National Ecological Observatory Network (NEON)](https://www.neonscience.org/) for the "Ground beetles sampled from pitfall traps" data product (DP1.10022.001, [RELEASE-2025](https://doi.org/10.48443/cd21-q875)) to predict drought conditions at the location and time of sampling. 

Reviewer: Yes, "carabid beetle" here refers to the family Carabidae (which the Wikipedia link is for), not some NEON-sampled subset or bespoke definition of the family.

> This challenge uses images and metadata associated with [carabid beetle](https://en.wikipedia.org/wiki/Ground_beetle) specimens collected by the [National Ecological Observatory Network (NEON)](https://www.neonscience.org/) for the "Ground beetles sampled from pitfall traps" data product (DP1.10022.001, [RELEASE-2025](https://doi.org/10.48443/cd21-q875)) to predict drought conditions at the location and time of sampling. 

Reviewer: But the NEON page could be added in the context of the NEON data product, because it also describes the trap sampling method.
